interfaces/cython/cantera/reaction.pyx

```diff
@@ -119,19 +439,69 @@ cdef class Reaction:
         cxx_reaction = CxxNewReaction(deref(CxxGetXmlFromString(stringify(text))))
         return Reaction.wrap(cxx_reaction)
 
-    @staticmethod
-    def fromYaml(text, Kinetics kinetics):
+    @classmethod
+    def from_dict(cls, data, Kinetics kinetics=None):
+        """
+        Create a `Reaction` object from a dictionary corresponding to its YAML
+        representation.
+
+        An example for the creation of a Reaction from a dictionary is::
+
+            rxn = Reaction.from_dict(
+                {"equation": "O + H2 <=> H + OH",
+                 "rate-constant": {"A": 38.7, "b": 2.7, "Ea": 26191840.0}},
+                kinetics=gas)
+
+        In the example, *gas* is a Kinetics (or Solution) object.
+
+        :param data:
+            A dictionary corresponding to the YAML representation.
+        :param kinetics:
+            A `Kinetics` object whose associated phase(s) contain the species
+            involved in the reaction.
+        """
+        if cls._reaction_type != "":
+            raise TypeError(
+                "Class method 'from_dict' was invoked from '{}' but should "
+                "be called from base class 'Reaction'".format(cls.__name__))
+        if kinetics is None:
+            raise ValueError("A Kinetics object is required.")
+
+        cdef CxxAnyMap any_map = dict_to_anymap(data)
+        cxx_reaction = CxxNewReaction(any_map, deref(kinetics.kinetics))
+        return Reaction.wrap(cxx_reaction)
+
+    @classmethod
+    def fromYaml(cls, text, Kinetics kinetics=None):
         """
         Create a `Reaction` object from its YAML string representation.
 
+        An example for the creation of a Reaction from a YAML string is::
+
+            rxn = Reaction.fromYaml('''
+                equation: O + H2 <=> H + OH
+                rate-constant: {A: 38.7, b: 2.7, Ea: 6260.0 cal/mol}
+                ''', kinetics=gas)
+
+        In the example, *gas* is a Kinetics (or Solution) object.
+
         :param text:
-            The YAML reaction string
+            The YAML reaction string; as an alternative, equivalent data can be
+            provided as a dictionary.
```

Comment:
This docstring is out of date now, right? A dictionary can only be used with `Reaction.from_dict`.